### DISCUSSION TITLE
GPU memory usage increases after every epoch for wav2vec2 model

### DISCUSSION THREAD

**Discussion by weiwei-ww** [speechbrain/speechbrain#2153] [Q&A]
I am using the wav2vec2 model from `speechbrain.lobes.models.huggingface_wav2vec.HuggingFaceWav2Vec2`. However, I found that the GPU memory usage kept increasing for the first few epochs. Here is the output of `torch.cuda.max_memory_allocated()` after the first few epochs.

After epoch 1: 16178949120

After epoch 2: 16989992448

After epoch 3: 17303541248

After epoch 6: 17461233152

After epoch 17: 17618925056

Afterwards, the GPU memory usage stays stable and no longer increases. Therefore, I do not think this is caused by memory leakage, but still, I am curious about what might be the cause of it?

**Comment by Gastron** [comment]
I am not too familiar with PyTorch memory management nor the Wav2Vec2 implementation, so this is a shot in the dark, but:

`max_memory_allocated` returns the peak memory usage since the start of the program - could it just be that the longer you run a program, the more likely it is to run into a very large batch, which records a new peak in memory usage? Perhaps you could try running [reset_peak_memory_stats](https://pytorch.org/docs/stable/generated/torch.cuda.reset_peak_memory_stats.html#torch.cuda.reset_peak_memory_stats) every epoch after checking the max_memory_allocated, so that you could see if the peak memory taken in each epoch just varies randomly?
